src/Graphics/OpenGLContext/GLSL/glsl_CombinerProgramBuilder.cpp

```diff
@@ -1016,15 +1016,30 @@ class ShaderFragmentHeaderReadTex : public ShaderPart
 					shaderPart +=
 						"#define TEX_OFFSET(off, tex, texCoord) texture(tex, texCoord - (off)/texSize)									\n"
 						"#define TEX_FILTER(name, tex, texCoord)												\\\n"
-						"  {																					\\\n"
+						"{																						\\\n"
 						"  mediump vec2 texSize = vec2(textureSize(tex,0));										\\\n"
 						"  mediump vec2 offset = fract(texCoord*texSize - vec2(0.5));							\\\n"
 						"  offset -= step(1.0, offset.x + offset.y);											\\\n"
 						"  lowp vec4 c0 = TEX_OFFSET(offset, tex, texCoord);									\\\n"
 						"  lowp vec4 c1 = TEX_OFFSET(vec2(offset.x - sign(offset.x), offset.y), tex, texCoord);	\\\n"
 						"  lowp vec4 c2 = TEX_OFFSET(vec2(offset.x, offset.y - sign(offset.y)), tex, texCoord);	\\\n"
+						"																						\\\n"
+						"  bool multiplyAlpha = uEnableAlphaTest == 1 || uCvgXAlpha == 1;						\\\n"
+						"  if(multiplyAlpha){																	\\\n" // Calculate premultiplied color values
+						"    c0.rgb = c0.rgb * vec3(c0.a);														\\\n"
+						"    c1.rgb = c1.rgb * vec3(c1.a);														\\\n"
+						"    c2.rgb = c2.rgb * vec3(c2.a);														\\\n"
+						"  }																					\\\n"
+						"																						\\\n"
 						"  name = c0 + abs(offset.x)*(c1-c0) + abs(offset.y)*(c2-c0); 							\\\n"
-						"  }																					\n"
+						"																						\\\n"
+						"  name.rgb = multiplyAlpha ? name.rgb / vec3(name.a) : name.rgb;						\\\n" // Divide alpha to get actual color value
```

Comment:
IIRC, use of conditional operators in shaders are bad for performance, if condition uses value, calculated in the shader. If condition uses only uniforms, it is ok. Thus, please replace
if(multiplyAlpha){
by
if(uEnableAlphaTest == 1 || uCvgXAlpha == 1){
and all code below must be moved to this if.

That is 
if (uEnableAlphaTest == 1 || uCvgXAlpha == 1) {
  c0.rgb = c0.rgb * vec3(c0.a);
...
  name = c0 + abs(offset.x)*(c1-c0) + abs(offset.y)*(c2-c0);
  if(name.a != 0.0 && name.a != 1.0){
...
  }
} else
  name = c0 + abs(offset.x)*(c1-c0) + abs(offset.y)*(c2-c0);

The same is for 4points bilinear.